scripts/count_flops.py

```diff
@@ -17,278 +17,31 @@
-            handle = module.register_forward_hook(pool_flops_counter_hook)
-        elif isinstance(module, torch.nn.BatchNorm2d):
-            handle = module.register_forward_hook(bn_flops_counter_hook)
-        else:
-            handle = module.register_forward_hook(empty_flops_counter_hook)
-        module.__flops_handle__ = handle
-
-
-def remove_flops_counter_hook_function(module):
-    """Internal auxiliary function"""
-    if is_supported_instance(module):
-        if hasattr(module, '__flops_handle__'):
-            module.__flops_handle__.remove()
-            del module.__flops_handle__
-
-
-def add_flops_mask_variable_or_reset(module):
-    """Internal auxiliary function"""
-    if is_supported_instance(module):
-        module.__mask__ = None
-
-
-def flops_to_string(flops):
-    """Converts flops count to a human-readable form"""
-    flops_str = ''
-    if flops // 10 ** 9 > 0:
-        flops_str = str(round(flops / 10. ** 9, 2)) + 'GMac'
-    elif flops // 10 ** 6 > 0:
-        flops_str = str(round(flops / 10. ** 6, 2)) + 'MMac'
-    elif flops // 10 ** 3 > 0:
-        flops_str = str(round(flops / 10. ** 3, 2)) + 'KMac'
-    else:
-        flops_str = str(flops) + 'Mac'
-    return flops_str
+from ptflops import get_model_complexity_info
 
 
 def main():
     """Runs flops counter"""
     parser = argparse.ArgumentParser(description='Evaluation script for Face Recognition in PyTorch')
     parser.add_argument('--embed_size', type=int, default=128, help='Size of the face embedding.')
-    parser.add_argument('--device', type=int, default=0, help='Device to store the model.')
     parser.add_argument('--model', choices=list(models_backbones.keys()) + list(models_landmarks.keys()), type=str,
                         default='rmnet')
     args = parser.parse_args()
 
-    with torch.cuda.device(args.device), torch.no_grad():
-        bs = 1
+    with torch.no_grad():
         if args.model in models_landmarks.keys():
-            model = add_flops_counting_methods(models_landmarks[args.model]())
-            batch = torch.Tensor(bs, 3, *model.get_input_res())
+            model = models_landmarks[args.model]()
         else:
-            net = models_backbones[args.model](embedding_size=args.embed_size, feature=True)
-            batch = torch.Tensor(bs, 3, *net.get_input_res())
-            model = add_flops_counting_methods(net)
+            model = models_backbones[args.model](embedding_size=args.embed_size, feature=True)
+        batch = torch.Tensor(1, 3, *model.get_input_res())
 
-        model.cuda().eval().start_flops_count()
-        output = model(batch.cuda())
+        output = model(batch)
+        flops, params = get_model_complexity_info(model, model.get_input_res(),
+                                                  as_strings=True, print_per_layer_stat=True)
 
-        print(model)
         print('Output shape: {}'.format(list(output.shape)))
```

Comment:
We already know the embedding size, so I'd prefer to remove this output.

Reply:
@sovrasov that's true. Removed.